src/core/a-entity.js

```diff
@@ -278,7 +278,11 @@ var proto = Object.create(ANode.prototype, {
   initComponent: {
     value: function (name, data, isDependency) {
       var component;
+      var componentInfo = name.split('__');
+      var id = componentInfo[1];
```

Comment:
umm, maybe `instanceId` ? and the `componentId` could be the whole name `componentName__instanceId`?